packages/scripts/utils/config.js

```diff
@@ -367,20 +377,29 @@ function getRenderPropPaths() {
 
 	const srcDirectory = fromProjectRoot( getWordPressSrcDirectory() + sep );
 
-	const renderPaths = blockMetadataFiles.map( ( blockMetadataFile ) => {
-		const { render } = JSON.parse( readFileSync( blockMetadataFile ) );
-		if ( render && render.startsWith( 'file:' ) ) {
+	return blockMetadataFiles.flatMap( ( blockMetadataFile ) => {
```

Comment:
Asking from ignorance, do we need to use `flatMap` instead of `map`? We are always dealing with strings after [this check](https://github.com/WordPress/gutenberg/pull/63098/files#diff-db38fb177aab60f25e5ed4bfc8b5ce6d371b2e17d96a465a04d89dab56aa8736R386), right? I don't know the implications it could have.

Reply:
Previously, we were dealing with one loop: We iterated over all `block.json` files, and from each of them, we extracted the `render` file (if there was one; otherwise, we set the path to `false`; we actually had to remove those values at the end, which is what the `renderPaths.filter( ( renderPath ) => renderPath );` was for).

But now, there's a second, nested loop: In each `block.json` file, we iterate over all props that could contain PHP filenames (currently `render` and `variations`), and if they're present, we keep track of their values. So at the `block.json` file level, we get an array that's either empty (if neither a `render` nor a `variations` field are found), or contains one element (if one of those fields are present), or two (if both are present).

This means that we end up with an array of arrays, which we then need to flatten. This is what the `flatMap` is for. (A welcome side effect is that we no longer need the `filter`, as this is now handled by flattening the empty arrays that we're getting if neither field is present in a `block.json` file).

Reply:
Thanks a lot for taking the time to provide the context 🙂